library/src/interactor_impl.cxx

```diff
@@ -268,6 +282,88 @@ class interactor_impl::internals
     self->Window.render();
   }
 
+  static void OnMiddleButtonPress(vtkObject*, unsigned long, void* clientData, void*)
+  {
+    internals* self = static_cast<internals*>(clientData);
+
+    const auto click_pos = self->VTKInteractor->GetEventPosition();
+    self->MiddleClickPosision[0] = click_pos[0];
+    self->MiddleClickPosision[1] = click_pos[1];
+
+    self->Style->OnMiddleButtonDown();
+  }
+
+  static void OnMiddleButtonRelease(vtkObject*, unsigned long, void* clientData, void*)
+  {
+    internals* self = static_cast<internals*>(clientData);
+
+    const auto click_pos = self->VTKInteractor->GetEventPosition();
+
+    const auto x_delta = click_pos[0] - self->MiddleClickPosision[0];
```

Comment:
Can you get the last position using `self->VTKInteractor->GetLastEventPosition();`? If it works, it would simplify the code.

Reply:
It seems `GetLastEventPosition` gets updated while dragging. If the pointer has stopped moving before releasing the button the values are the same as `GetEventPosition`.